### PR TITLE
evm: add forge deployment script

### PR DESCRIPTION
**Pull request by gator-boi**

### CODE REVIEW

File: evm/script/ConfigureWormholeNtt.s.sol [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                    targetConfig.chainId, toUniversalAddress(targetConfig.wormholeTransceiver)
+                );
+                console2.log("Wormhole peer set for chain", targetConfig.chainId);
+
+                // Set EVM chain.
+                if (targetConfig.isEvmChain) {
+                    wormholeTransceiver.setIsWormholeEvmChain(targetConfig.chainId);
+                    console2.log("EVM chain set for chain", targetConfig.chainId);
+                } else {
+                   revert("Non-EVM chain is not supported yet");
```
Comment (djb15) [inline evm/script/ConfigureWormholeNtt.s.sol:109]: What about Solana?

File: evm/script/ConfigureWormholeNtt.s.sol [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                continue;
+            } else {
+                // Set peer.
+                nttManager.setPeer(
+                    targetConfig.chainId, toUniversalAddress(targetConfig.nttManager), targetConfig.decimals
+                );
+                console2.log("Peer set for chain", targetConfig.chainId);
+
+                // Configure the inbound limit.
+                nttManager.setInboundLimit(targetConfig.inboundLimit, targetConfig.chainId);
```
Comment (djb15) [inline evm/script/ConfigureWormholeNtt.s.sol:128]: So by default every other chain will have the same inbound limit coming from a given chain?

**Review by djb15 — COMMENTED** [review]
This is cool!

Will approve once the rate limit skipping PR has landed as the script will have to change slightly to accept the extra constructor

File: evm/script/ConfigureWormholeNtt.s.sol (hunk 1 above)
Comment (gator-boi) [reply evm/script/ConfigureWormholeNtt.s.sol:109]: Happy to add Solana before we merge this.

File: evm/script/ConfigureWormholeNtt.s.sol (hunk 2 above)
Comment (gator-boi) [reply evm/script/ConfigureWormholeNtt.s.sol:128]: This script is executed once for each chain, so the integrator could modify the inbound limit for each chain that it's registering.

File: evm/script/DeployWormholeNtt.s.sol [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    function configureNttManger(
+        address nttManager,
+        address transceiver,
+        uint64 outboundLimit
+    ) public {
+        INttManager(nttManager).setTransceiver(transceiver);
+        console2.log("Transceiver address set on NttManager: ", transceiver);
+
+        INttManager(nttManager).setOutboundLimit(outboundLimit);
```
Comment (djb15) [inline evm/script/DeployWormholeNtt.s.sol:85]: Could skip this is the rate limiter is disabled

File: evm/script/ConfigureWormholeNtt.s.sol [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                continue;
+            } else {
+                // Set peer.
+                nttManager.setPeer(
+                    targetConfig.chainId, targetConfig.nttManager, targetConfig.decimals
+                );
+                console2.log("Peer set for chain", targetConfig.chainId);
+
+                // Configure the inbound limit.
+                nttManager.setInboundLimit(targetConfig.inboundLimit, targetConfig.chainId);
```
Comment (djb15) [inline evm/script/ConfigureWormholeNtt.s.sol:80]: Same, could skip this if rate limiting is disabled

**Review by solanoepalacio — DISMISSED** [review]

File: evm/README.md [hunk 5]
```diff
@@ -166,9 +180,66 @@ $ make fix-fmt
...
+
+### Deploy Wormhole NTT
+
+#### Environment Setup
+
+Copy the sample environment file located in `env/` into the target subdirectory of your choice (currently `testnet` or `mainnet`) and prefix the filename with your blockchain of choice:
+
+```
+cp env/.env.sample env/testnet/sepolia.env
+```
```
Comment (RahulMaganti47) [inline evm/README.md:205]: add a note here to create the `testnet` dir before running this command

File: evm/README.md [hunk 6]
```diff
@@ -166,9 +180,66 @@ $ make fix-fmt
...
+
+Do this for each blockchain network that the `NTTManager` and `WormholeTransceiver` contracts will be deployed to. Then configure each `.env` file and set the `RPC` variables.
+
+#### Config Setup
+
+Before deploying the contracts, navigate to the `cfg` directory and copy the sample file. Make sure to preserve the existing name:
+
```
Comment (RahulMaganti47) [inline evm/README.md:212]: add note to navigate back to the `evm` dir

**Review by RahulMaganti47 — COMMENTED** [review]
some readme additions

File: evm/README.md (hunk 5 above)
Comment (gator-boi) [reply evm/README.md:205]: Whoops, I meant to create that ahead of time actually.

Comment (kcsongor) [reply evm/README.md:205]: better, include a mkdir command just before the cp step

Comment (gator-boi) [reply evm/README.md:205]: Sweet, that works.

File: evm/script/DeployWormholeNtt.s.sol [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+            params.token, params.mode, params.wormholeChainId, params.rateLimitDuration
+        );
+
+        // NttManager Proxy
+        NttManager nttManagerProxy =
+            NttManager(address(new ERC1967Proxy(address(implementation), "")));
+
+        nttManagerProxy.initialize();
+
+        console2.log("NttManager deployed at: ", address(nttManagerProxy));
```
Comment (RahulMaganti47) [inline evm/script/DeployWormholeNtt.s.sol:49]: log address in 32-byte wormhole format

File: evm/script/DeployWormholeNtt.s.sol [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+            params.wormholeRelayerAddr,
+            params.specialRelayerAddr,
+            params.consistencyLevel,
+            params.gasLimit
+        );
+
+        ERC1967Proxy transceiver = new ERC1967Proxy(address(wormholeTransceiver), "");
+        WormholeTransceiver(address(transceiver)).initialize();
+
+        console2.log("Wormhole Transceiver deployed at: ", address(wormholeTransceiver));
```
Comment (RahulMaganti47) [inline evm/script/DeployWormholeNtt.s.sol]: same as above - log in 32-byte format

File: evm/script/DeployWormholeNtt.s.sol [hunk 9]
```diff
@@ -62,23 +59,29 @@ contract DeployWormholeNtt is Script {
...
     }
 
-    /// @notice Sets the Wormhole Transceiver address and outbound rate limiter limit on the NttManager.
     function configureNttManger(
         address nttManager,
         address transceiver,
         uint64 outboundLimit
     ) public {
         INttManager(nttManager).setTransceiver(transceiver);
+        console2.log("Transceiver address set on NttManager: ", transceiver);
```
Comment (RahulMaganti47) [inline evm/script/DeployWormholeNtt.s.sol:85]: log in 32-byte format

Comment (gator-boi) [reply evm/script/DeployWormholeNtt.s.sol:88]: fixed

**Review by RahulMaganti47 — APPROVED** [review]

**Review by nik-suri — APPROVED** [review]

**Pull request merged**